script_parameters.yml

```diff
@@ -0,0 +1,238 @@
+          echo "my_integer = $MY_INTEGER"
+          echo "my_float = $MY_FLOAT"
+          # To prevent command injection attacks, you can get shell-escaped versions of string parameters.
+          echo "my_string = $MY_STRING_SHELL_ESCAPED"
+          echo "my_file is at $MY_FILE_URL"
+          # It is generally not advisable to log your credentials.
+          # my_custom_credential generates $MY_CUSTOM_CREDENTIAL_USERNAME, $MY_CUSTOM_CREDENTIAL_PASSWORD
+          # my_aws_credential generates $MY_AWS_CREDENTIAL_ACCESS_KEY_ID, $MY_AWS_CREDENTIAL_SECRET_ACCESS_KEY
+          # my_db_credential generates $MY_DB_CREDENTIAL_USERNAME, $MY_DB_CREDENTIAL_PASSWORD
+        # You can add this if you want to prevent the created script from
+        # appearing on the scripts index page, to reduce clutter.
+        hidden: true
+      on-success:
+        - python_script
+
+    python_script:
+      action: civis.scripts.python3
+      input:
+        required_resources:
+          cpu: 256
+          memory: 512
+          disk_space: 1
+        params:
+          # Optionally, you can specify the parameter values in the `params` list
+          # instead of in the `arguments` hash, by using the `default` key.  But
+          # this does not work for credential parameters.
+          - name: my_integer
+            type: integer
+            default: 1234
+          - name: my_float
+            type: float
+            default: 0.1234
+          - name: my_string
+            type: string
+            default: this is a string
+          - name: my_file
+            type: file
+            default: 0  # ID of the file
+          - name: my_custom_credential
+            type: credential_custom
+          - name: my_aws_credential
+            type: credential_aws
+          - name: my_db_credential
+            type: credential_redshift
+        arguments:
+          my_custom_credential: 0  # ID of the credential
+          my_aws_credential: 0
+          my_db_credential: 0
+        source: |
+          # Parameter values are added to your container as environment variables.
+          # See the script params reference for a full list.
+
+          import os
+
+          # Environment variables are retrieved as strings, so if you want to use
+          # them as numbers, you'll need to do type conversion.
+          my_integer = int(os.environ['MY_INTEGER'])
+          my_float = float(os.environ['MY_FLOAT'])
+
+          my_string = os.environ['MY_STRING']
+          my_file_url = os.environ['MY_FILE_URL']
+
+          my_custom_credential = (os.environ['MY_CUSTOM_CREDENTIAL_USERNAME'], os.environ['MY_CUSTOM_CREDENTIAL_PASSWORD'])
+          my_aws_credential = (os.environ['MY_AWS_CREDENTIAL_ACCESS_KEY_ID'], os.environ['MY_AWS_CREDENTIAL_SECRET_ACCESS_KEY'])
+          my_db_credential = (os.environ['MY_DB_CREDENTIAL_USERNAME'], os.environ['MY_DB_CREDENTIAL_PASSWORD'])
+      on-success:
+        - r_script
+
+    r_script:
```

Comment:
Something to note here is that these are names, but a beginner might be confused and thing this is a way to specify the script type. Even changing these to "my_r_script" and "my_python_script" might help avoid that simple confusion